File: 37.MemoryDemo/37.MemoryDemo/ViewController.m
//
//  ViewController.m
//  37.MemoryDemo
//
//  Created by jackfrow on 2020/5/27.
//  Copyright © 2020 jackfrow. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

- (IBAction)Allocate:(id)sender {
    
    for (int i = 0; i < 1000; i++) {
        Size* s  =  malloc(1024*512);
        s[0] = 1;
        free(s);
    }
    
}


@end

File: 26.ProtocolDemo/ProtocolDemo/抖动/ViewController.swift
//
//  ViewController.swift
//  ProtocolDemo
//
//  Created by jackfrow on 2019/10/10.
//  Copyright © 2019 jackfrow. All rights reserved.
//

import UIKit



class ViewController: UIViewController {
    
    
    @IBOutlet weak var image: FoodImageView!
    @IBOutlet weak var ActionBtn: ActionButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }

    @IBAction func takeAshake(_ sender: Any) {
        
        image.shake()
        ActionBtn.shake()
        
    }
    
}

